File: src/core/services/settings_schedule_changes.rs
use crate::core::models::AppSettings;

pub struct SettingsScheduleChanges;

impl SettingsScheduleChanges {
    pub fn eye_schedule_changed(old: &AppSettings, new: &AppSettings) -> bool {
        old.eye_reset_enabled != new.eye_reset_enabled
            || old.eye_reset_interval_minutes != new.eye_reset_interval_minutes
            || old.eye_reset_duration_seconds != new.eye_reset_duration_seconds
    }

    pub fn move_schedule_changed(old: &AppSettings, new: &AppSettings) -> bool {
        old.move_break_enabled != new.move_break_enabled
            || old.move_break_interval_minutes != new.move_break_interval_minutes
            || old.move_break_duration_seconds != new.move_break_duration_seconds
    }
}

File: برنامه ویندوزی/AWAYRA-WPF-main (2)/src/Awayra.Core/Services/SettingsScheduleChanges.cs
using Awayra.Core.Models;

namespace Awayra.Core.Services;

public static class SettingsScheduleChanges
{
    public static bool EyeScheduleChanged(AppSettings original, AppSettings updated) =>
        original.EyeResetEnabled != updated.EyeResetEnabled ||
        original.EyeResetIntervalMinutes != updated.EyeResetIntervalMinutes ||
        original.EyeResetDurationSeconds != updated.EyeResetDurationSeconds;

    public static bool MoveScheduleChanged(AppSettings original, AppSettings updated) =>
        original.MoveBreakEnabled != updated.MoveBreakEnabled ||
        original.MoveBreakIntervalMinutes != updated.MoveBreakIntervalMinutes ||
        original.MoveBreakDurationSeconds != updated.MoveBreakDurationSeconds;
}
